### PR TITLE
Fail tx sanitization when ix program id uses lookup table

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Transaction ingestion cannot do static analysis over transaction program instructions to determine user fees, precompile usage, and estimated compute without first loading addresses from on-chain lookup tables.

#### Summary of Changes
- Update transaction sanitization to require that tx instructions use static program ids

Fixes #
Feature Gate Issue: https://github.com/solana-labs/solana/issues/25034

### CODE REVIEW

File: sdk/program/src/message/versions/v0/mod.rs [hunk 1]
```diff
@@ -97,37 +102,63 @@ impl Sanitize for Message {
...
-        // encoded as `u8`
-        if num_loaded_accounts > 256 {
+        // the combined number of static and dynamic account keys must be <= 256
+        // since account indices are encoded as `u8`
+        let total_account_keys = num_static_account_keys.saturating_add(num_dynamic_account_keys);
+        if total_account_keys > 256 {
             return Err(SanitizeError::IndexOutOfBounds);
         }
 
+        let max_account_ix = total_account_keys
```
Comment (t-nelson) [inline sdk/program/src/message/versions/v0/mod.rs:137]:
```suggestion
        // `expect` is safe here due to prior check that `num_static_account_keys` is
        // non-zero
        let max_account_ix = total_account_keys
```

File: sdk/program/src/message/versions/v0/mod.rs [hunk 2]
```diff
@@ -97,37 +102,63 @@ impl Sanitize for Message {
...
+            .checked_sub(1)
+            .expect("message doesn't contain any account keys");
+
+        // switch to rejecting program ids loaded from lookup tables so that
+        // static analysis on program instructions can be performed without
+        // loading on-chain data from a bank
+        let max_program_id_ix = if reject_dynamic_program_ids {
+            num_static_account_keys
```
Comment (t-nelson) [inline sdk/program/src/message/versions/v0/mod.rs:145]:
```suggestion
            // `expect` is safe here due to prior check that `num_static_account_keys`
            // is non-zero
            num_static_account_keys
```

**Review by t-nelson — APPROVED** [review]
lgtm, just nits

File: sdk/program/src/message/versions/mod.rs [hunk 3]
```diff
@@ -148,15 +155,6 @@ impl Default for VersionedMessage {
     }
 }
 
-impl Sanitize for VersionedMessage {
```
[the remark this answers is not in the document]
Comment (jstarry) [reply sdk/program/src/message/versions/mod.rs:151]: Yeah unfortunate...

**Pull request merged**